### PR TITLE
Feature: print QR options

### PR DESCRIPTION
**Pull request by andi34**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/andi34/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)

- [x] New feature

#### What changes did you make? (Give an overview)

New settings while printing QR-Code onto the image:
- QR size adjustable
- QR offset adjustable
- QR position adjustable
- QR error correction adjustable

By default place the QR-Code on the bottom right corner, use an small QR-Code with an small offset of 10px to the edges.

#### Is there anything you'd like reviewers to focus on?
Screenshot_20220125-232740

Screenshot_20220125-232752

Screenshot_20220125-232758

### CODE REVIEW

File: api/print.php [hunk 1]
```diff
@@ -74,27 +82,52 @@
                 $url = $config['qr']['url'];
             }
             include '../vendor/phpqrcode/lib/full/qrlib.php';
-            QRcode::png($url, $filename_codes, QR_ECLEVEL_H, 10);
+            QRcode::png($url, $filename_codes, QR_ECLEVEL_H, $config['print']['qrSize']);
+            if ($rotateQr) {
+                $image = imagecreatefrompng($filename_codes);
+                $resultRotated = imagerotate($image, 90, 0); // Rotate image
+                imagepng($resultRotated, $filename_codes, 0);
+                imagedestroy($image); // Destroy the created collage in memory
```
Comment (andi34) [inline api/print.php]: Not a collage, fix the comment

File: api/print.php [hunk 2]
```diff
@@ -74,27 +82,52 @@
                 $url = $config['qr']['url'];
             }
             include '../vendor/phpqrcode/lib/full/qrlib.php';
-            QRcode::png($url, $filename_codes, QR_ECLEVEL_H, 10);
+            QRcode::png($url, $filename_codes, QR_ECLEVEL_H, $config['print']['qrSize']);
```
Comment (andi34) [inline api/print.php]: Make the ECC adjustable too?

**Review by andi34 — COMMENTED** [review]
@up-87 what do you think?

File: api/print.php (hunk 2 above)
Comment (up-87) [reply api/print.php]: Depends on how much impact it has. I didn't play around with it by now.

**Review by up-87 — APPROVED** [review]

File: api/print.php (hunk 2 above)
Comment (andi34) [reply api/print.php]: Not much, another switch case would need to be added. There's 4 ECC options we could use. Plus the config stuff for the Adminpanel.

Comment (up-87) [reply api/print.php]: In this case I'd stay away for it. But maybe choose a different level anyway? https://www.qrcode.com/en/about/error_correction.html
> Level Q or H may be selected for factory environment where QR Code get dirty

I don't feel like this is the typical use case or requirement for a photobooth...
https://blog.qrstuff.com/2011/12/14/qr-code-error-correction

> Level L or Level M represent the best compromise between density and ruggedness for general marketing use.

Comment (andi34) [reply api/print.php]: There might be different use cases which why I'd consider to add a switch and let the user decide, but sure we can get down to a lower profile by default while on it.

Comment (andi34) [reply api/print.php]: Might be enough to set $config['print']['qrEcc'] and make it a select list inside Adminpanel, so no need for a switch case

Comment (andi34) [reply api/print.php]: Or maybe in general for QR and move this setting to QR section. Same applies to the qr inside gallery

**Pull request merged**